src/main/java/com/thealgorithms/backtracking/MazeRecursion.java

```diff
@@ -1,66 +1,76 @@
 package com.thealgorithms.backtracking;
 
+/**
+ * This class contains methods to solve a maze using recursive backtracking.
+ * The maze is represented as a 2D array where walls, paths, and visited/dead
+ * ends
+ * are marked with different integers.
+ *
+ * The goal is to find a path from a starting position to the target position
+ * (map[6][5]) while navigating through the maze.
+ */
 public final class MazeRecursion {
+
     private MazeRecursion() {
     }
 
+    /**
+     * This method sets up a maze as a 2D array, where '1' represents walls and '0'
+     * represents open paths. It then calls recursive functions to find paths using
+     * two different movement strategies. The results are printed to the console.
+     */
     public static void mazeRecursion() {
```

Comment:
To improve this class, please make methods accept maze to solve as a parameter and return the solution instead of printing it. Also add JUnit tests

Reply:
Change implemented!